pkg/model/spotinstmodel/instance_group.go

```diff
@@ -343,22 +343,25 @@ func (b *InstanceGroupModelBuilder) buildOcean(c *fi.ModelBuilderContext, igs ..
 									InstanceGroupLabelOceanDefaultLaunchSpec)
 							}
 
-							ig = g
+							ig = g.DeepCopy()
 							break
 						}
 					}
 				}
 			}
+
+			// No default instance group. Use the first one.
 			if ig == nil {
-				return fmt.Errorf("unable to detect default launch spec: "+
-					"please label the desired default instance group with `%s: \"true\"`",
-					InstanceGroupLabelOceanDefaultLaunchSpec)
+				ig = igs[0].DeepCopy()
 			}
 		}
 
 		klog.V(4).Infof("Detected default launch spec: %q", b.AutoscalingGroupName(ig))
 	}
 
+	// Rename the instance group to avoid duplicate tasks with same name.
+	ig.Name = fi.StringValue(ocean.Name)
```

Comment:
HEy @liranp !
Could it be that this line introduced a bug I might be facing?

By overriding ig.Name it messes up userdata for default nodes:
ocean.UserData, err = b.BootstrapScriptBuilder.ResourceNodeUp(c, ig)

Because now nodeUp tries to fetch instance group config with name "ocean.Name" which ofcourse it doesn;t exist

Reply:
@povils, thanks for reporting this. I've raised another PR (#10630) to fix it. Sorry about that.